src/Providers.js

```diff
@@ -31,6 +33,6 @@ class Providers extends PureComponent {
 }
 
 export default connect(state => ({
-  locale: state.settings.locale,
-  theme: state.settings.theme,
+  locale: getLocale(state),
```

Comment:
Extracting direct state access to a selector is generally OK. On one hand it might be a bit of an over-kill, but on the other hand, it abstracts the location we get this data from, so we can move it more freely.